cpp/src/parquet/properties.h

```diff
@@ -116,6 +116,12 @@ class PARQUET_EXPORT ReaderProperties {
     page_checksum_verification_ = check_crc;
   }
 
+  bool use_binary_large_variants() const { return use_binary_large_variants_; }
```

Comment:
It would be moved to `ArrowReaderProperties` to advise the return type of arrow binary arrays. 

To be clear, `ReaderProperties` controls the behavior the internal parquet reader (without any arrow dependency) while `ArrowReaderProperties` controls the arrow-related behaviors around parquet.